types/react/index.d.ts

```diff
@@ -107,30 +125,16 @@ declare namespace React {
      * Represents any user-defined component, either as a function or a class.
      *
      * Similar to {@link ComponentType}, but without extra properties like
-     * {@link FunctionComponent.defaultProps defaultProps } and
-     * {@link ComponentClass.contextTypes contextTypes}.
+     * {@link FunctionComponent.defaultProps defaultProps }.
      *
      * @template P The props the component accepts.
      */
     type JSXElementConstructor<P> =
         | ((
             props: P,
-            /**
-             * @deprecated
-             *
-             * @see {@link https://legacy.reactjs.org/docs/legacy-context.html#referencing-context-in-stateless-function-components React Docs}
-             */
-            deprecatedLegacyContext?: any,
         ) => ReactNode)
-        | (new(
-            props: P,
-            /**
-             * @deprecated
-             *
-             * @see {@link https://legacy.reactjs.org/docs/legacy-context.html#referencing-context-in-lifecycle-methods React Docs}
-             */
-            deprecatedLegacyContext?: any,
-        ) => Component<any, any>);
+        // constructor signature must match React.Component
+        | (new(props: P) => Component<any, any>);
 
     /**
      * A readonly ref container where {@link current} cannot be mutated.
```

Comment:
This description should be updated now that the interface is not readonly, right?